src/utils/load_theme.cpp

```diff
@@ -18,203 +18,164 @@
-    ELSE_COLOR(DARK_RED)
-    ELSE_COLOR(MEDIUM_RED)
-    ELSE_COLOR(PINK)
-
-    ELSE_COLOR(GREEN)
-    ELSE_COLOR(WABGREEN)
-    ELSE_COLOR(WABGREEN_BIS)
-    ELSE_COLOR(DARK_WABGREEN)
-    ELSE_COLOR(INV_DARK_WABGREEN)
-    ELSE_COLOR(DARK_GREEN)
-    ELSE_COLOR(INV_DARK_GREEN)
-    ELSE_COLOR(LIGHT_GREEN)
-    ELSE_COLOR(INV_LIGHT_GREEN)
-    ELSE_COLOR(PALE_GREEN)
-    ELSE_COLOR(INV_PALE_GREEN)
-    ELSE_COLOR(MEDIUM_GREEN)
-    ELSE_COLOR(INV_MEDIUM_GREEN)
 
-    ELSE_COLOR(YELLOW)
-    ELSE_COLOR(LIGHT_YELLOW)
-
-    ELSE_COLOR(ORANGE)
-    ELSE_COLOR(LIGHT_ORANGE)
-    ELSE_COLOR(PALE_ORANGE)
-    ELSE_COLOR(BROWN)
-#undef ELSE_COLOR
-    else if ((*str == '0') && (*(str + 1) == 'x')){
-        bgr = BGRasRGBColor(BGRColor(strtol(str + 2, nullptr, 16)));
-    }
-    else {
-        bgr = BGRasRGBColor(BGRColor(strtol(str, nullptr, 10)));
-    }
-
-    return bgr;
-}
-
-
-// 1, yes, on, true
-static bool bool_from_cstr(const char * value)
+namespace
 {
-    bool val;
-    auto err = configs::parse(val, configs::spec_type<bool>(), array_view{value, strlen(value)});
-    return err ? false : val;
-}
-
-
-struct ThemeHolder final : public ConfigurationHolder
-{
-    explicit ThemeHolder(Theme & theme)
-    : theme(theme)
-    {}
-
-    void set_value(const char * context, const char * key, const char * value) override
+    inline std::pair<bool, long int>
+    to_long_int_base16_or_10(std::string_view str_view) noexcept
     {
-        if (0 == strcmp(context, "global")) {
-            if (0 == strcmp(key, "bgcolor")){
-                this->theme.global.bgcolor = color_from_cstr(value);
-            }
-            else if (0 == strcmp(key, "fgcolor")){
-                this->theme.global.fgcolor = color_from_cstr(value);
-            }
-            else if (0 == strcmp(key, "separator_color")){
-                this->theme.global.separator_color = color_from_cstr(value);
-            }
-            else if (0 == strcmp(key, "focus_color")){
-                this->theme.global.focus_color = color_from_cstr(value);
-            }
-            else if (0 == strcmp(key, "error_color")){
-                this->theme.global.error_color = color_from_cstr(value);
-            }
-            else if (0 == strcmp(key, "logo")){
-                this->theme.global.logo = bool_from_cstr(value);
-            }
+        std::pair<bool, long int> res { false, 0 };
+        
+        if (str_view.empty())
+            return res;
+        
+        auto converter = [](const char *str, int base) -> decltype(res)
+        {
+            long int value = 0;
+            bool success = false;
+            char *endptr = nullptr;
+
+            errno = 0;
+            value = strtol(str, &endptr, base);
+            success = !(endptr == str || *endptr != '\0' || errno);
+            return { success, value };
+        };
+
+        if (str_view.size() > 2
+            && *str_view.cbegin() == '0'
+            && *(str_view.cbegin() + 1) == 'x')
```

Comment:
`str_view[0] == '0' && str_view[1] == 'x'`

Reply:
I had a little forgotten the std::string_view::operator[] ...

Changed !